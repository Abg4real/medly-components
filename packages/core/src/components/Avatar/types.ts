import { AvatarSizes } from '@medly-components/theme';
import { WithThemeProp } from '@medly-components/utils';

export interface Props extends WithThemeProp {
    size?: keyof AvatarSizes;
    textColor?: string;
    bgColor?: string;
}

export interface StyledProps extends Props {
    fontSize: string;
    avatarSize: string;
}
