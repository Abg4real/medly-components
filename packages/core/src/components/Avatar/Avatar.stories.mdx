import { Avatar } from './Avatar.tsx';
import { Preview, Story, Meta, Props } from '@storybook/addon-docs/blocks';
import * as stories from './Avatar.stories';

<Meta title="Core/Avatar" component={Avatar} parameters={{ jest: ['Avatar.test.tsx'] }} />

# Avatar

### Displays an image or text that represents a user or organization

<Preview withToolbar>
    <Story name="Basic">{stories.Basic()}</Story>
</Preview>

### Props

<Props of={Avatar} />

### AvatarTheme

<Props of={stories.ThemeInterface} />
