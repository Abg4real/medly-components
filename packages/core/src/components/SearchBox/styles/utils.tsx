import { SvgIcon } from '@medly-components/icons';
import { css } from '@medly-components/utils';
import { Props } from '../types';

export const getSearchBoxSize = ({ theme, size }: Props) => theme.searchBox.sizes[size].height;

export const getBorderAndBoxShadow = (color: string, boxShadow: string) => {
    return css`
        border-color: ${color};
        box-shadow: ${boxShadow};
    `;
};

export const getIconWrapperStyle = ({ theme, size }: Props) => {
    return css`
        width: ${getSearchBoxSize};
        min-width: ${getSearchBoxSize};
        height: auto;
        display: inline-flex;
        align-items: center;
        justify-content: center;
        ${SvgIcon} {
            cursor: pointer;
            font-size: ${theme.searchBox.sizes[size].iconSize};
        }
    `;
};
