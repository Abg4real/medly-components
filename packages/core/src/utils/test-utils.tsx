import { defaultTheme } from '@medly-components/theme';
import React from 'react';
import { render, RenderResult } from 'react-testing-library';
import { ThemeProvider } from './styled';

const AllTheProviders: React.FunctionComponent = props => (
    <ThemeProvider theme={defaultTheme}>
        <>{props.children}</>
    </ThemeProvider>
);

const customRender = (ui: React.ReactElement<any>, options?: object): RenderResult => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from 'react-testing-library';

// override render method
export { customRender as render };
