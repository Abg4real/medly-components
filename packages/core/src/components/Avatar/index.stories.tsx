import { color, select } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';
import Avatar from './Avatar';
import { Props } from './types';

const size: Array<Props['size']> = ['S', 'M', 'L', 'XL'];

storiesOf('Core', module).add('Avatar', () => (
    <>
        <Avatar size={select('Size', size, 'M')}>
            <img src="http://i.pravatar.cc/150?img=66" />
        </Avatar>
        <Avatar size={select('Size', size, 'M')} bgColor={color('Background', 'aqua')} textColor={color('Text', 'white')}>
            M
        </Avatar>
    </>
));
