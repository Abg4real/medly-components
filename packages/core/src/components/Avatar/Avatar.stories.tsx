import { AvatarTheme, defaultTheme } from '@medly-components/theme';
import { styled } from '@medly-components/utils';
import { color, select } from '@storybook/addon-knobs';
import React from 'react';
import { Avatar } from './Avatar';
import { Props } from './types';

const size: Props['size'][] = ['S', 'M', 'L', 'XL'];

export const ThemeInterface: React.SFC<AvatarTheme> = () => null;
ThemeInterface.defaultProps = {
    ...defaultTheme.avatar
};

const Div = styled.div`
    & > * {
        margin-right: 10px;
    }
`;

export const Basic = () => (
    <Div>
        <Avatar size={select('Size', size, 'M')}>
            <img src="http://i.pravatar.cc/150?img=66" />
        </Avatar>
        <Avatar size={select('Size', size, 'M')} bgColor={color('Background', 'aqua')} textColor={color('Text', 'white')}>
            M
        </Avatar>
    </Div>
);
