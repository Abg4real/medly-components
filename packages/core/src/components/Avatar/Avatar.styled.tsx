import { css, styled } from '@medly-components/utils';
import Text from '../Text';
import { StyledProps } from './types';

const getAvatarSize = ({ theme: { avatar }, size }: StyledProps) => {
    const { sizes, defaults } = avatar;
    return sizes[size || defaults.size].avatarSize;
};

const getShadowColor = ({ theme: { avatar }, hoverTextShadowColor, hoverImgShadowColor, isImage }: StyledProps) => {
    return isImage
        ? hoverImgShadowColor || avatar.defaults.hover.imgShadowColor
        : hoverTextShadowColor || avatar.defaults.hover.textShadowColor;
};

const getTextStyle = ({ theme, ...props }: StyledProps) => {
    const { size, textColor, bgColor } = props,
        { defaults, sizes } = theme.avatar,
        { avatarSize, fontSize } = sizes[size || defaults.size];

    return css`
        color: ${textColor || defaults.textColor};
        background: ${bgColor || defaults.bgColor};
        user-select: none;

        ${Text.Style} {
            line-height: ${avatarSize};
            font-size: ${fontSize};
            font-weight: ${theme.font.weights[defaults.fontWeight]};
            font-family: ${defaults.fontFamily};
        }
    `;
};

export const AvatarStyled = styled('div').attrs(({ theme: { avatar: { defaults } } }: StyledProps) => ({
    defaults
}))<StyledProps>`
    display: inline-block;
    text-align: center;
    width: ${getAvatarSize};
    height: ${getAvatarSize};
    border-radius: 50%;
    overflow: hidden;
    cursor: ${({ onClick, disabled }) => (disabled ? 'not-allowed' : onClick ? 'pointer' : 'inherit')};

    ${getTextStyle}

    img {
        width: ${getAvatarSize};
        height: ${getAvatarSize};
        object-fit: cover;
        border: 0.1rem solid ${({ defaults }) => defaults.borderColor};
        box-sizing: border-box;
        border-radius: 50%;
    }

    &:hover {
        background: ${({ defaults, hoverBgColor }) => hoverBgColor || defaults.hover.bgColor};
        box-shadow: 0 0.4rem 0.8rem ${getShadowColor};
        ${Text.Style} {
            color: ${({ defaults, hoverTextColor }) => hoverTextColor || defaults.hover.textColor};
        }
    }
`;
