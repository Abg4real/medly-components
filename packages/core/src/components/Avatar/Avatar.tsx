import { isValidStringOrNumber, WithStyle } from '@medly-components/utils';
import React, { SFC } from 'react';
import Text from '../Text';
import { AvatarStyled } from './Avatar.styled';
import { Props } from './types';

export const Avatar: SFC<Props> & WithStyle = React.memo(
    React.forwardRef((props, ref) => {
        return (
            <AvatarStyled ref={ref} {...props}>
                {React.Children.map(props.children, c => {
                    return isValidStringOrNumber(c) ? (
                        <Text uppercase textWeight="Strong">
                            {c}
                        </Text>
                    ) : (
                        c
                    );
                })}
            </AvatarStyled>
        );
    })
);

Avatar.displayName = 'Avatar';
Avatar.Style = AvatarStyled;
