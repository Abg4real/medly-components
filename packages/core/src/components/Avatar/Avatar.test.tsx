import { TestUtils } from '@medly-components/utils';
import React from 'react';
import Avatar from './Avatar';

describe('Avatar component', () => {
    it('should render with default props', () => {
        const { container } = TestUtils.render(<Avatar>M</Avatar>);
        expect(container).toMatchSnapshot();
    });

    it('should render with size', () => {
        const { container } = TestUtils.render(<Avatar size="M">M</Avatar>);
        expect(container).toMatchSnapshot();
    });

    it('should render with size and image', () => {
        const { container } = TestUtils.render(
            <Avatar size="XL">
                <img src="http://dummurl" />
            </Avatar>
        );
        expect(container).toMatchSnapshot();
    });
});
