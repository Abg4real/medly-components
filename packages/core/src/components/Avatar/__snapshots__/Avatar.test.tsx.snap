// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component should render with default props 1`] = `
.c2 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 700;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-transform: uppercase;
}

.c0 {
  display: inline-block;
  text-align: center;
  width: 40px;
  height: 40px;
  border-radius: 50%;
  overflow: hidden;
  color: #005AEE;
  background: #E6EFFD;
}

.c0 .c1 {
  line-height: 40px;
  font-size: 16px;
  font-weight: normal;
}

.c0 img {
  width: 40px;
  height: 40px;
  object-fit: cover;
  border: 0.1rem solid #DFE4E9;
  box-sizing: border-box;
  border-radius: 50%;
}

.c0 img:hover {
  box-shadow: 0 0.4rem 0.8rem #607890;
}

.c0:hover {
  box-shadow: 0 0.4rem 0.8rem #005aee;
  background: #1A6BF0;
}

.c0:hover .c1 {
  color: #FFFFFF;
}

<div>
  <div
    class="c0"
  >
    <strong
      class="c1 c2"
    >
      M
    </strong>
  </div>
</div>
`;

exports[`Avatar component should render with size 1`] = `
.c2 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 700;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-transform: uppercase;
}

.c0 {
  display: inline-block;
  text-align: center;
  width: 40px;
  height: 40px;
  border-radius: 50%;
  overflow: hidden;
  color: #005AEE;
  background: #E6EFFD;
}

.c0 .c1 {
  line-height: 40px;
  font-size: 16px;
  font-weight: normal;
}

.c0 img {
  width: 40px;
  height: 40px;
  object-fit: cover;
  border: 0.1rem solid #DFE4E9;
  box-sizing: border-box;
  border-radius: 50%;
}

.c0 img:hover {
  box-shadow: 0 0.4rem 0.8rem #607890;
}

.c0:hover {
  box-shadow: 0 0.4rem 0.8rem #005aee;
  background: #1A6BF0;
}

.c0:hover .c1 {
  color: #FFFFFF;
}

<div>
  <div
    class="c0"
  >
    <strong
      class="c1 c2"
    >
      M
    </strong>
  </div>
</div>
`;

exports[`Avatar component should render with size and image 1`] = `
.c0 {
  display: inline-block;
  text-align: center;
  width: 48px;
  height: 48px;
  border-radius: 50%;
  overflow: hidden;
  color: #005AEE;
  background: #E6EFFD;
}

.c0 .sc-AykKC {
  line-height: 48px;
  font-size: 20px;
  font-weight: normal;
}

.c0 img {
  width: 48px;
  height: 48px;
  object-fit: cover;
  border: 0.1rem solid #DFE4E9;
  box-sizing: border-box;
  border-radius: 50%;
}

.c0 img:hover {
  box-shadow: 0 0.4rem 0.8rem #607890;
}

.c0:hover {
  box-shadow: 0 0.4rem 0.8rem #607890;
  background: #1A6BF0;
}

.c0:hover .sc-AykKC {
  color: #FFFFFF;
}

<div>
  <div
    class="c0"
  >
    <img
      src="http://dummurl"
    />
  </div>
</div>
`;
